### DISCUSSION TITLE
[Feature] Collapsible project files on Repository overview

### DISCUSSION THREAD

**Discussion by MAZ01001** [community/community#109986] [Repositories]
### Select Topic Area

General

### Body

# Feature request: Collapsable project files

To better or rather quicker see the README for a Repository, especially those with many files, the project files list can be collapsed into a small button, something like this.

<details><summary>Expanded (current look)</summary>

![normal](https://github.com/community/community/assets/29178226/ba563bcd-d515-49c1-8a58-b1f084b0f541 "Screenshot of a Repository")

</details>

<details><summary>Collapsed (modified HTML)</summary>

![collapsed](https://github.com/community/community/assets/29178226/cd6c61ee-b5fe-4802-ab34-545d19cd09ad "Screenshot of a Repository with the project files collapsed (modified HTML)")
![collapsed_hover](https://github.com/community/community/assets/29178226/bcbad716-2ccb-4258-9279-78ec3b9df672 "Screenshot of a Repository with the project files collapsed (modified HTML) while hovering over the \"Click to expand\" button")

</details>

Every Repository with a README should have the project files default collapsed.

A Repository setting like `show project files [auto|expanded|collapsed]` should also be added.

**Comment by x9136** [comment]
+.
#112978

**Comment by MAZ01001** [comment]
in the mean time use this: [MAZ01001/CollapsibleGitHubProjectFiles](https://github.com/MAZ01001/CollapsibleGitHubProjectFiles "[GitHub] MAZ01001/CollapsibleGitHubProjectFiles") (tampermonkey userscript)

**Reply by x9136** [reply]
Thanks! You may want to contribute to [the issue](https://github.com/refined-github/refined-github/issues/6959) so that this feature is returned to the Refined GitHub extension.

**Reply by MAZ01001** [reply]
_The userscript now also shows the number of rows._

**Reply by MAZ01001** [reply]
_and now has more options to customize default collapse state_

**Reply by x9136** [reply]
I prefer to stay on the first version. 👍 The rest lead to this:
![Screenshot 2024-05-10 at 16 50 04](https://github.com/community/community/assets/60105060/e117593c-fc0d-426e-90b1-986a7b9c9136)

**Reply by MAZ01001** [reply]
Try v2.2 now @x9136 I made some improvements and it should be a little faster...

**Comment by GalenReich** [comment]
The addition of a repository setting is a great idea. Particularly for users with less familiarity of the GitHub UI, getting to the README as quickly as possible can really improve their experience.

For me the default could remain as is, but the addition of a `show project files [auto|expanded|collapsed]` repository setting would be a helpful feature. 

Users should probably have the ability to override the behavior in the repository setting with a user setting.
